tracer/test/Datadog.Trace.IntegrationTests/StatsTests.cs

```diff
@@ -568,7 +568,7 @@ void AssertTraces(IReadOnlyList<MockSpan> payload, bool expectStats)
                 // occurrence would be considered "rare". However, the RareSampler
                 // only runs after the PrioritySampler, so assert the metric "_dd.rare"
                 // based on the SamplingPriority
-                bool expectRareMetric = expectStats && payload[0].GetMetric(Metrics.SamplingPriority) <= 0;
+                bool expectRareMetric = expectStats && payload[0].GetMetric(Metrics.SamplingPriority) != (int)SamplingPriority.UserReject;
```

Comment:
The `SamplingPriority` enum is deprecated in favor of `SamplingPriorityValues`. We couldn't remove it entirely because it's part of public APIs.
```suggestion
                bool expectRareMetric = expectStats && payload[0].GetMetric(Metrics.SamplingPriority) != SamplingPriorityValues.UserReject;
```

Reply:
Also, this changes the condition from `<= 0` to `<= -1`. Was that intentional?

Reply:
Yeah, with this change the rare sampler also evaluates the span if the priority is 0

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/3360/commits/e47895afa5fb745eeb754955307a65fae9db10df